lib/PrometheusPlugin.js

```diff
@@ -0,0 +1,244 @@
+        buckets: Prometheus.exponentialBuckets(0.0001, 1.5, 36),
+        labelNames: ['controller', 'action', 'event', 'status', 'protocol']
+      }),
+      rooms: new Prometheus.Gauge({
+        name: `kuzzle_active_rooms`,
+        help: `Kuzzle active rooms counter`
+      })
+    };
+
+    this.hooks = {
+      'room:new': this.recordRooms.bind(this),
+      'room:remove': this.recordRooms.bind(this),
+      'request:onSuccess': this.recordRequests.bind(this),
+      'request:onError': this.recordRequests.bind(this)
+    };
+
+    this.systemMetricsJob = Prometheus.collectDefaultMetrics({
+      register: this.register,
+      timeout: this.config.systemMetricsInterval || 5000
+    });
+
+    this.syncJob = setInterval(
+      () => this.pushRegistry(),
+      this.config.syncInterval || 7500
+    );
+  }
+
+  /**
+   * Hooks
+   */
+
+  /**
+   * Record rooms creation to Prometheus Registry
+   * when 'room:new' Kuzzle Event is triggered.
+   *
+   * @param {Object} _
+   * @param {String} event
+   */
+  async recordRooms(_, event) {
+    switch (event) {
+      case 'room:new':
+        this.kuzzleMetrics.rooms.inc();
+        break;
+      case 'room:remove':
+        this.kuzzleMetrics.rooms.dec();
+        break;
+    }
+  }
+
+  /**
+   * Record requests information to Prometheus Registry.
+   *
+   * @param {Request} request
+   * @param {String}  event
+   */
+  async recordRequests(request, event) {
+    if (request.input.action != 'metrics') {
+      this.kuzzleMetrics.requests.observe(
+        {
+          controller: request.input.controller,
+          action: request.input.action,
+          event,
+          protocol: request.context.connection.protocol
+        },
+        Date.now() - request.timestamp
+      );
+    }
+  }
+
+  /**
+   * Route
+   */
+
+  /**
+   * Custom route to fetch all saved metrics
+   * in Prometheus metrics format
+   *
+   * @param {Request} request
+   */
+  async metrics(request) {
+    const register = await this.syncRegisters();
+    request.response.setHeader('Content-Type', register.contentType);
+    request.response.raw = true;
+    return register.metrics();
+  }
+
+  /**
+   * Helpers
+   */
+
+  /**
+   * Push given Prometheus registers as string
+   * to Redis using the provided key
+   */
+  async pushRegistry() {
+    await this.context.accessors.sdk.ms.set(
+      this.key,
+      JSON.stringify(this.registry.getMetricsAsJSON())
+    );
+  }
+
+  /**
+   * Pull Prometheus registers stored in Redis merge them
+   * with the local one and return a new Registry which
+   * contains all Prometheus metrics
+   *
+   * @return {Prometheus.Registry}
+   */
+  async syncRegisters() {
+    let cursor = 0,
+      values = [];
+
+    do {
+      let result = await this.context.accessors.sdk.ms.scan(cursor, {
+        match: 'kuzzle_prometheus_*'
+      });
+
+      cursor = result.cursor;
+      values = values.concat(result.values);
+    } while (cursor != 0);
```

Comment:
```suggestion
    } while (cursor !== 0);
```

Reply:
I know but this will not work, cursor is a string :slightly_smiling_face:

Reply:
`} while (cursor !== '0');`